dev-libs/hyperscan/hyperscan-5.4.0.ebuild

```diff
@@ -0,0 +1,54 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6..9} )
+
+inherit cmake flag-o-matic python-any-r1
+
+DESCRIPTION="High-performance regular expression matching library"
+SRC_URI="https://github.com/intel/hyperscan/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="https://www.hyperscan.io/ https://github.com/intel/hyperscan"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="cpu_flags_x86_avx2 +cpu_flags_x86_ssse3 static-libs"
+
+RDEPEND="dev-libs/boost"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-util/ragel
+"
+
+REQUIRED_USE="cpu_flags_x86_ssse3"
+
+src_prepare() {
+	# Respect user -O flags
+	sed -i '/set(OPT_CX*_FLAG/d' CMakeLists.txt || die
+
+	# upstream workaround
+	append-cxxflags -Wno-redundant-move
+	cmake_src_prepare
+}
+
+src_configure() {
+	CMAKE_BUILD_TYPE=Release
```

Comment:
We shouldn't need to set this - the cmake eclass does it.

Reply:
The eclass sets `-DCMAKE_BUILD_TYPE=Gentoo`, but `hyperscan` cmake matches it with `Release`. This varibale is necessary, see https://github.com/gentoo/gentoo/pull/15681#discussion_r436924060, where I introduce it. Anyway, maybe I should add some comment to it.

Reply:
Oh, great. Thanks. I'll restore it with a comment.

Reply:
`-Werror` is not the only thing that is triggered by `Release` build type. Maybe, the best will be to patch `CMakeList.txt` this way:
```diff
@@ -30,7 +30,7 @@ else()
     message(STATUS "Build type ${CMAKE_BUILD_TYPE}")
 endif()

-if(CMAKE_BUILD_TYPE MATCHES RELEASE|RELWITHDEBINFO|MINSIZEREL)
+if(CMAKE_BUILD_TYPE MATCHES RELEASE|RELWITHDEBINFO|MINSIZEREL|Gentoo)
     message(STATUS "using release build")
     set(RELEASE_BUILD TRUE)
 else()
```

Reply:
Btw, I just have checked that Gentoo does same `CMakeFile.txt` fixes with `CMAKE_BUILD_TYPE` set to `Gentoo` or `Release`.